internal/internal_task_handlers.go

```diff
@@ -616,6 +616,10 @@ func (wth *workflowTaskHandlerImpl) ProcessWorkflowTask(
 
 	workflowContext, err := wth.getOrCreateWorkflowContext(task, historyIterator)
 	if err != nil {
+		// The WorkflowContext will be removed from the cache if an error is found
+		// in workflowContext.err when calling Unlock() method.
+		workflowContext.err = err
```

Comment:
Have you tested this? I don't think this would work. Even if the workflow type is not defined, the getOrCreateWorkflowContext() won't return error. The error happen when it try to process the WorkflowExecutionStartedEvent.
Also, we should not try to set workflowContext.err directly.

Reply:
yeah, I was somehow thought the error comes from getOrCreateWorkflowContext and mocked an error and tested on that. 

Gotcha, I won't set the workflowContext.err directly.